OpenRA.Game/Exts.cs

```diff
@@ -113,19 +113,25 @@ public static int IndexOf<T>(this T[] array, T value)
 
 		public static T Random<T>(this IEnumerable<T> ts, MersenneTwister r)
 		{
-			var xs = ts as ICollection<T>;
-			if (xs != null)
-				return xs.ElementAt(r.Next(xs.Count));
-			var ys = ts.ToList();
-			return ys[r.Next(ys.Count)];
+			return Random(ts, r, true);
 		}
 
 		public static T RandomOrDefault<T>(this IEnumerable<T> ts, MersenneTwister r)
 		{
-			if (!ts.Any())
-				return default(T);
+			return Random(ts, r, false);
+		}
 
-			return ts.Random(r);
+		static T Random<T>(IEnumerable<T> ts, MersenneTwister r, bool throws)
+		{
+			var xs = ts as ICollection<T>;
+			xs = xs ?? ts.ToList();
```

Comment:
Aye of course you are correct.
I made the mistake of not reading the new code below this comment and was mentally referring to the indexing we had done beforehand.